src/csharp/Microsoft.Spark.E2ETest/SparkSettings.cs

```diff
@@ -35,7 +35,13 @@ private static void InitVersion()
             // Spark 2.3.2 built for Hadoop 2.7.3
             string firstLine =
                 File.ReadLines($"{SparkHome}{Path.DirectorySeparatorChar}RELEASE").First();
-            Version = new Version(firstLine.Split(' ')[1]);
+
+            // Grab "2.3.2" from "Spark 2.3.2 built for Hadoop 2.7.3"
+            string versionStr = firstLine.Split(' ')[1];
+
+            // Strip anything below version number.
+            // For example, "3.0.0-preview" should become "3.0.0".
+            Version = new Version(versionStr.Split('-')[0]);
```

Comment:
This is technically not needed, but I didn't remove this b/c it could be useful when testing preview versions.